### ISSUE TITLE
[Feature Request] Delay and cancel all syncing on launch of the app

### ISSUE THREAD

**Issue by ankushdharkar** [simplerocket-llc/dear-cashew#43]
Being able to pause the syncing process when starting the app for the first time is becoming very crucial (for me)

I have added a lot of repos, and they try and sync everytime I launch the app. **The app stays unresponsive and shows a blank screen**. I don't even know if the repos are actually being synced at all, since I don't know which repo is being synced _now_ and how much of it is left. Something like #37 will help

Something like `Cancel Sync` on the top bar, letting me cancel the syncing process in the first 10 seconds of running the app can really make the experience better. You can also provide a `Sync now` button (which is already kind of there as the refresh button in the bottom of the sidebar)

**Comment by hishboy** [comment]
How many repos do you synced so far? do you mind listing the public ones?

**Comment by ankushdharkar** [comment]
Can't count, the app won't show me (blank screen, unresponsive)
The big one which took a really long time: [ember-cli/ember-cli](https://github.com/ember-cli/ember-cli/issues)

Some others:
`[ember-cli/rfcs](https://github.com/ember-cli/rfcs/issues)`
`[ember-cli/ember-cli-eslint](https://github.com/ember-cli/ember-cli-eslint/issues)`
`[ember-cli/loader.js](https://github.com/ember-cli/loader.js/issues)`

**Comment by ankushdharkar** [comment]
If you like to be daring, add this: 

[emberjs/ember.js](https://github.com/emberjs/ember.js/issues)

**Comment by hishboy** [comment]
I want to do address this issue in stages. Sounds like the biggest issue here is syncing too many repositories at once. 

Some background on this. When the app starts, I start a delta syncer which syncs all repositories since the last time the app was open. I then run a delta syncer every 2 times just to  minimize large syncs. The app startup syncer seems to be causing the freezes for you. Another thing to note is the syncer downloads issue details along with its supplementary data i.e. events and comments. This can be quite expensive operation. 

I’m planning to disable pulling down suppliementary data and just request it when the user clicks on the issue for `v1.7`. I’ve experienced much better results with this approach. 

In the future, I’m thinking about allowing the user to pick which repositories they would like to access offline vs only online. This will minimize the number of syncs as well. Also, I really like the idea of pausing syncs for repositories users don’t really care about at the moment. I’ll take care of that in `v1.8`

**Comment by ankushdharkar** [comment]
Nice
> I’m planning to disable pulling down suppliementary data and just request it when the user clicks on the issue

This is cool 
> I’m thinking about allowing the user to pick which repositories they would like to access offline vs only online

**Comment by ankushdharkar** [comment]
How do I solve the mess I am in. Even if I delete the app and its contents, when I download a new one, it will sync the repo list from iCloud and start syncing them

**Comment by hishboy** [comment]
Let me see if I can manually delete the repos for you from iCloud. Are they added under @ankushdharkar?

**Comment by ankushdharkar** [comment]
Yes. Thank you!

**Comment by hishboy** [comment]
K, try now

**Comment by ankushdharkar** [comment]
Still loading everything. I am going to delete the contents now and logging in

**Comment by ankushdharkar** [comment]
Alright, that worked. Thank you so much @hishboy! 🙌

**Comment by hishboy** [comment]
👍🏼

**Comment by hishboy** [comment]
@ankushdharkar alot of fixes & performance enhancements coming to `1.8`. I just submitted it to the App Store but it should really help with synching performance issues and app startup.  Still not 100% and much better improvement.

**Comment by ankushdharkar** [comment]
@hisboy: Awesome! Will it still break the app if I add repos with 1k+ issues? Or is it doing the online fetchin on demand you had mentioned earlier?

**Comment by hishboy** [comment]
Yes, the app should work fine with repos larger than 1k+. I actually have multiple repos @ 20K+ including ember.js 😊. You just have to give it some time to sync on the first full pass.

Bottom line, the performace of the app should not degrade with more repos or issues. At least that’s my goal. If it is, please let me know.

Not doing the on demand fetching yet.

**Comment by ankushdharkar** [comment]
@hishboy: Cool. I actually removed a lot of big repos from Cashew because it slowed things down. Will add them back in `1.8`. Keep up the good work! 🎉
